raiden/transfer/mediated_transfer/state.py

```diff
@@ -55,6 +55,26 @@ def __init__(self, our_address, transfer, routes, block_number, random_generator
         self.revealsecret = None
         self.canceled_transfers = list()
 
+    def __eq__(self, other):
+        if isinstance(other, InitiatorState):
```

Comment:
You have changed the code since the review and used this `safe_equal_attributes()` function and check each attribute manually. As such my previous comment is no longer relevant.